File: strategy/en_us/python/src/celsius_to_fahrenheit.py
"""Converter: A pseudo-Interface for the strategies"""
from converter import Converter

class CelsiusToFahrenheit(Converter):
	"""
	Class responsible to convert a measure in Celsius into Fahrenheit.
	It is a Concrete Strategy of the context.
	"""

	CELSIUS = "*C (Celsius)"
	FAHRENHEIT = "*F (Fahrenheit)"

	def convert(self, current_measure):
		"""
		Convert a measure in Celsius into Fahrenheit.
		For more information:
			https://en.wikipedia.org/wiki/Conversion_of_units_of_temperature
		"""

		converted_measure = (current_measure * (9.0/5.0)) + 32.0

		print "%.3f %s = %.3f %s" % (current_measure, self.CELSIUS,
			converted_measure, self.FAHRENHEIT)

		return converted_measure

File: strategy/en_us/java/CelsiusToFahrenheit.java
/**
 *	File: CelsiusToFahrenheit
 *  Role: Concrete Strategy
 *  Purpose: Class responsible to convert a measure in Celsius into Fahrenheit.
 */
public class CelsiusToFahrenheit implements Converter {


	/**
	 *	Convert a measure in Celsius into Fahrenheit.
	 *
	 *	@param currentMeasure	a measure in Celsius.
	 *	@return	the measure converted into Fahrenheit
	 */
	@Override
	public float convert(float currentMeasure) {

		/*
		 *	Converting Celsius to Fahrenheit.
		 *	For more information:
		 *		https://en.wikipedia.org/wiki/Conversion_of_units_of_temperature
		 */
		float convertedMeasure = (currentMeasure * (9f / 5f)) + 32f;

		/*
		 *	Inform the conversion as follows: <X>°C (Celsius) = <Y>°F (Fahrenheit).
		 *	Used to show a result of an application using Strategy Pattern.
		 */
		System.out.println(currentMeasure + CELSIUS + " = " + convertedMeasure + FAHRENHEIT);

		return convertedMeasure;
	}
}
